### PR TITLE
Seeds

### PR DESCRIPTION
**Pull request by amunozo**

### CODE REVIEW

File: train/train_params.py [hunk 1]
```diff
@@ -25,14 +25,8 @@ class TrainParams:
...
     early_stop_nsamples: int = -1
-
-
-def make_tp(forgiving=False, takes_extras=False, convert_lists_to_tuples=False,
-            **d):
-    return apply_dataclass(TrainParams, d, forgiving=forgiving,
-                           convert_lists_to_tuples=convert_lists_to_tuples)
-    # ready for fixes over time
-
+    weight_decay: float = 0.01
```
Comment (gailweiss) [inline train/train_params.py:28]: stray - weight decay shouldn't be in the random seed commit

File: train/train_params.py [hunk 2]
```diff
@@ -109,4 +103,8 @@ def make_tp(forgiving=False, takes_extras=False, convert_lists_to_tuples=False,
 #   Relevant only when also passing --save through the args, otherwise will be
 #   overridden to 0.
 # early_stop_nsamples:
-#   When >0, the number of batches to train before early stopping the training.
\ No newline at end of file
+#   When >0, the number of batches to train before early stopping the training.
+# weight_decay:
+#   Weight decay to apply to the optimizer.
```
Comment (gailweiss) [inline train/train_params.py:108]: stray - weight decay shouldnt be in the random seed commit

File: train/train_params.py [hunk 3]
```diff
@@ -109,4 +103,8 @@ def make_tp(forgiving=False, takes_extras=False, convert_lists_to_tuples=False,
...
+# random_seed:
+#   Random seed for reproducibility
```
Comment (gailweiss) [inline train/train_params.py:110]: maybe add a note here that this value can also be overwritten by setting the random seed arg in main

**Comment by amunozo** [comment]
yes, i tested it on several runs with the test-nlp config and result where exactly the same.

I think I mixed some of the commits when creating this branch that's why there is some weight decay stuff. I will delete it and upload it here.

File: train/train_params.py [hunk 4]
```diff
@@ -25,7 +25,6 @@ class TrainParams:
     lora_std: float = 0.02
     checkpoint_every: int = 0
     early_stop_nsamples: int = -1
-    weight_decay: float = 0.01
     random_seed: None = None
```
Comment (gailweiss) [inline train/train_params.py:28]: this should be random_seed: int = None

**Pull request merged**